### ISSUE TITLE
Is it possible to add a custom loss function or modify the current one? 

### ISSUE THREAD

**Issue by AmeetR** [DeepLabCut/DeepLabCut#655]
I'd like to add a few task specific terms to the loss function to ensure that some things that I know should be true with correct predictions. Is it possible to modify the loss function without editing source code and recompiling the entire library?

**Comment by AlexEMG** [comment]
Can you perhaps describe what type of terms you want to add to the loss?

**Comment by AmeetR** [comment]
Assuming that we're training sequentially and we have close to all the frames in a video, I was thinking it might be possible to add a component to the loss function that penalizes the prediction at time t-1 being too far away from time t. I'm hoping this gets rid of some noise in the labeling.

**Comment by AlexEMG** [comment]
Currently training is iid. If you want to change this, then yes, you would need to change the code & then reinstall.

**Comment by AmeetR** [comment]
oh ok, thanks -- I'll look into doing that (althought I would probably phase the training)
